build.sh

```diff
@@ -19,8 +20,15 @@ DRYRUN=
 SHOW_VERSION=false
 SCRIPT_ARGUMENTS=()
 
+# Run without mono on windows platforms.
+case "$(uname -s)" in
+    CYGWIN*) unset MONO_CMD ;;
+    MINGW32*) unset MONO_CMD ;;
+    MSYS*) unset MONO_CMD ;;
+esac
+
 # Parse arguments.
-for i in "$@"; do
+while [[ $# -gt 0 ]]; do
```

Comment:
Changing the evaluation here because on at least mingw the command `./build.sh --target Run-Unit-Tests` produces an empty string to be evaluated for `SCRIPT_ARGUMENTS`; ultimately causing `cake.exe` to think we are passing multiple scripts and fails.